### ISSUE TITLE
Retrying causes the cookies to be added multiple times

### ISSUE THREAD

**Issue by RonaldinhoL** [imroc/req#242]
```
func main() {
	client := req.C()

	client.SetUserAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36")

	client.R().Get("https://www.baidu.com")

	client.SetProxyURL("http://127.0.0.1:8888")

	client.SetCommonRetryCount(3)
	client.SetCommonRetryCondition(func(resp *req.Response, err error) bool {
		return true
	})

	client.R().Get("https://www.baidu.com")
}
```

**Comment by RonaldinhoL** [comment]
![微信截图_20230608112454](https://github.com/imroc/req/assets/1797392/904df362-2871-4d62-906c-e8b4b6047db6)

**Comment by imroc** [comment]
Fixed and released in v3.35.2
